### DISCUSSION TITLE
How to make strimzi support kafka 1.1.1 and 2.4.0

### DISCUSSION THREAD

**Discussion by yamicro** [strimzi/strimzi-kafka-operator#10068] [Q&A]
We currently have two versions of kafka clusters, 1.1.1 and 2.4.0. I currently want to build the strimzi development environment for both versions first, to get a deeper understanding while validating strimzi. When I read the documentation, I found that strimzi is able to support 2.4.0. However, when I build the project using mvn clean install -DskipTests -e after modifying the support field to true for 2.4.0 in kafka-verision.yaml, I get the following error:
````
[ERROR] Failed to execute goal on project config-model-generator: Could not resolve dependencies for project io.strimzi:config-model-generator:jar:0.41.0-SNAPSHOT: The following artifacts could not be resolved: org.apache.kafka:kafka-server-common:jar:2.4.0, org.apache.kafka:kafka-raft:jar:2.4.0: Could not find artifact org.apache.kafka:kafka-server-common:jar:2.4.0 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
````
I didn't find kafka-serve-common for 2.4.0 under https://repo.maven.apache.org/maven2, the oldest version of 3.0.0. I'm not quite sure what kafka-serve-common is used for, and where should I go to get the older version of kafka serve-common yet.
I hope you can help me with this, and let me know if I'm supporting versions 1.1.1 and 2.4.0 in the wrong way now.

**Comment by scholzj** [comment]
Kafka 2.4.0 and 1.1.1 are very old. I don't think anyone can tell you what will be the amount of changes needed to make it work. But my guess is that it would be significant because many APIs changed since those versions were released and you would need to modify the source code and not just some Maven dependencies. But I think nobody ever done it, so you are on your own to figure out how to do it.

**Reply by yamicro** [reply]
Okay I got it. Thanks a lot.
